dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry/internal/rest/TelemetryContainerFilter.java

```diff
@@ -176,7 +180,9 @@ public String route(final ContainerRequestContext request) {
                 String contextRoot = request.getUriInfo().getBaseUri().getPath();
                 UriBuilder template = UriBuilder.fromPath(contextRoot);
 
-                template.path(resourceClass);
+                if (resourceClass.isAnnotationPresent(Path.class)) {
+                    template.path(resourceClass);
+                }
```

Comment:
Hmm. I thought that if the `@Path` annotation wasn't on the class, then the class wouldn't be found and used as a resource class.

We need to raise an issue for this scenario so that we can ensure that the Telemetry code is doing the right thing here. At the moment the "fix" just skips part of the logic that computes the route.